### PR TITLE
Move ServiceBus extension to Azure SDK repo

### PR DESCRIPTION
**Pull request by JoshLove-msft**
https://github.com/Azure/azure-sdk-for-net/issues/17560
Moving the extensions from https://github.com/Azure/azure-functions-servicebus-extension with minimal changes necessary to compile and have tests passed. This will be helpful as a reference for future changes as we migrate to use the Track 2 library.

Notable issues that should be converted to GH issues:
- signing the sign.snk did not work as there was no private key, so I'm using the SDK repo snk - https://github.com/Azure/azure-sdk-for-net/issues/17958
- TestCommon package did not support net461 so I updated the RequiredTargetFrameworks for the test project to just be netcoreapp2.1 - https://github.com/Azure/azure-sdk-for-net/issues/17959

### CODE REVIEW

**Review by jsquire — APPROVED** [review]

**Comment by pakrym** [comment]
Should we delete the snk file if we are not using it?

**Review by pakrym — APPROVED** [review]

**Comment by jsquire** [comment]
My approval is based on the understanding that we're moving over existing assets;  I've only reviewed the new SDK repository assets like the CI configuration and Changelog in any depth.

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/stylecop.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿{
+  // ACTION REQUIRED: This file was automatically added to your project, but it
+  // will not take effect until additional steps are taken to enable it. See the
+  // following page for additional information:
+  //
+  // https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/EnableConfiguration.md
+
+  "$schema": "https://raw.githubusercontent.com/DotNetAnalyzers/StyleCopAnalyzers/master/StyleCop.Analyzers/StyleCop.Analyzers/Settings/stylecop.schema.json",
```
Comment (jsquire) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/stylecop.json:8]: Is this redundant to the one in `/api` ?

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/test/Microsoft.Azure.WebJobs.Extensions.ServiceBus.Tests/WebJobs.Extensions.ServiceBus.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <WarningsAsErrors />
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <StyleCopTreatErrorsAsWarnings>false</StyleCopTreatErrorsAsWarnings>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <WarningsAsErrors />
+  </PropertyGroup>
+  <ItemGroup>
+    <AdditionalFiles Remove="C:\Users\jolov\src\r\Azure\azure-sdk-for-net\sdk\servicebus\Microsoft.Azure.WebJobs.Extensions.ServiceBus\\stylecop.json" />
```
Comment (jsquire) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/test/Microsoft.Azure.WebJobs.Extensions.ServiceBus.Tests/WebJobs.Extensions.ServiceBus.Tests.csproj:23]: You've got a local path for your environment still in here...

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/test/Microsoft.Azure.WebJobs.Extensions.ServiceBus.Tests/WebJobs.Extensions.ServiceBus.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="2.1.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.5.0" />
+    <PackageReference Include="Moq" Version="4.7.145" />
+    <PackageReference Include="xunit" Version="2.3.1" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.3.1" />
+    <DotNetCliToolReference Include="dotnet-xunit" Version="2.3.0-beta3-build3705" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <AdditionalFiles Remove="C:\Users\jolov\src\r\Azure\azure-sdk-for-net\eng\stylecop.json" />
```
Comment (jsquire) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/test/Microsoft.Azure.WebJobs.Extensions.ServiceBus.Tests/WebJobs.Extensions.ServiceBus.Tests.csproj:42]: Local path for your machine here too...

**Comment by JoshLove-msft** [comment]
> Should we delete the snk file if we are not using it?

Leaving for now as we want to use this eventually.

**Comment by weshaggard** [comment]
Are the "snk" files in this PR needed? Is there a reason they are not using the standard snk files from the eng folder in the repo?

**Comment by JoshLove-msft** [comment]
> Are the "snk" files in this PR needed? Is there a reason they are not using the standard snk files from the eng folder in the repo?

No, they aren't, but we will add new ones once we get them from the service team. The reason for not using the standard ones is that we don't want to change the full assembly name between versions.

**Comment by weshaggard** [comment]
For the shipping libraries I think it is fine but for the test assemblies there is not reason to have a checked unique key. Also for the test folder I see both a sign.snk and the AzureSdkTestLibKey.snk file. I don't think there should be any reason to have either of those in the test folder.

**Pull request merged**